lib/mongo/cluster/app_metadata.rb

```diff
@@ -55,6 +55,11 @@ def initialize(cluster)
         @app_name = cluster.options[:app_name].to_s if cluster.options[:app_name]
         @platform = cluster.options[:platform]
         @compressors = cluster.options[:compressors] || []
+
+        if cluster.options[:user] && !cluster.options[:auth_mech]
+          auth_db = cluster.options[:auth_source] || 'admin'
```

Comment:
Isn't default auth database supposed to be the same as connection database?

Reply:
The authentication options for a connection will be the same as the cluster that its server belongs to, which is what's being checked here before applying the default.